File: software/examples/c/example_simple.c

#include <stdio.h>

#include "ip_connection.h"
#include "bricklet_temperature_ir.h"

#define HOST "localhost"
#define PORT 4223
#define UID "abcde" // Change to your UID

int main() {
	// Create IP connection to brickd
	IPConnection ipcon;
	if(ipcon_create(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not create connection\n");
		exit(1);
	}

	// Create device object
	TemperatureIR tir;
	temperature_ir_create(&tir, UID); 

	// Add device to IP connection
	if(ipcon_add_device(&ipcon, &tir) < 0) {
		fprintf(stderr, "Could not connect to Brick\n");
		exit(1);
	}
	// Don't use device before it is added to a connection

	// Get current ambient and object temperatures (unit is °C/10)
	uint16_t obj;
	uint16_t amb;
	temperature_ir_get_object_temperature(&tir, &obj);
	temperature_ir_get_ambient_temperature(&tir, &amb);

	printf("Object Temperature: %f °C\n", obj/10.0);
	printf("Ambient Temperature: %f °C\n", amb/10.0);

	printf("Press ctrl+c to close\n");
	ipcon_join_thread(&ipcon); // Join mainloop of IP connection
}

File: software/examples/php/ExampleSimple.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletTemperatureIR.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletTemperatureIR;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change to your UID

$ipcon = new IPConnection(); // Create IP connection
$tir = new BrickletTemperatureIR(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Get current ambient temperature (unit is °C/10)
$ambient_temperature = $tir->getAmbientTemperature();
echo "Ambient Temperature: " . $ambient_temperature/10.0 . " °C\n";

// Get current object temperature (unit is °C/10)
$object_temperature = $tir->getObjectTemperature();
echo "Object Temperature: " . $object_temperature/10.0 . " °C\n";

echo "Press key to exit\n";
fgetc(fopen('php://stdin', 'r'));
$ipcon->disconnect();

?>
